google/auth/identity_pool.py

```diff
@@ -216,6 +216,16 @@ def _parse_token_data(
             )
         return token
 
+    def _create_default_metrics_options(self):
+        metrics_options = super(Credentials, self)._create_default_metrics_options()
+        try:
+            if self._credential_source.get("file"):
```

Comment:
```suggestion
            if self._credential_source.get("file", None):
```

What is the context behind the use of a `try` block?

Reply:
Added the try because this method runs before the full __init__ completes for identity pool credentials since it gets called by the super().__init__. If the try isn't there, the error thrown comes from the metrics method instead of being the expected error. Will add a comment

Reply:
example of test that would fail as well - https://github.com/googleapis/google-auth-library-python/blob/49d0f9eeb7887e0cc1f3b2686cf03a2c2dc4a1db/tests/test__default.py#LL535C1-L544C6

Reply:
Is it possible to refactor the constructor to enforce this invariant before this function is called?

This approach makes the function very complex and tough to refactor. 

Perhaps this function should be refactored to be a property, so it is guaranteed that the object is fully instantiated?

Reply:
@clundin25 Why do we prefer an explicit `None` in your suggestion? I thought by default it's `None` already.

Reply:
My thought was that the try/catch could be replaced by an if statement. Now that I see it's a workaround for the interaction with the constructor.

I would prefer seeing the logic refactored to not need such a try/catch, but my initial code suggestion is no longer relevant.

Reply:
Its a bit difficult either way, _metrics_options is needed in the parent class constructor because the impersonated credential object is created there, and the _initialize_impersonated_credentials method needs to get the _metrics_options value, so even if we switch it to a property, it will run before the constructor finishes. Could move the super().__init__ to be after the credential source validation for identity pool but that seems a more significant change to fix this